openedx/core/djangoapps/waffle_utils/views.py

```diff
@@ -3,277 +3,124 @@
     """
-    Gets all waffle switches and their state.
+    Override some of the methods from ToggleStateReport to expose toggles from WaffleFlagCourseOverrideModel objects.
     """
-    switches_dict = {}
-    _add_waffle_switch_instances(switches_dict)
-    _add_waffle_switch_state(switches_dict)
-    _add_waffle_switch_computed_status(switches_dict)
-    switch_list = list(switches_dict.values())
-    switch_list.sort(key=lambda toggle: toggle['name'])
-    return switch_list
 
+    def add_waffle_flag_instances(self, flags_dict):
+        """
+        Append objects from WaffleFlagCourseOverrideModel.
+        """
+        super().add_waffle_flag_state(flags_dict)
+        _add_waffle_flag_course_override_state(flags_dict)
 
-def _add_waffle_switch_instances(switches_dict):
-    """
-    Add details from waffle switch instances, like code_owner.
-    """
-    waffle_switch_instances = WaffleSwitch.get_instances()
-    for switch_instance in waffle_switch_instances:
-        switch = _get_or_create_toggle_response(switches_dict, switch_instance.name)
-        _add_toggle_instance_details(switch, switch_instance)
+    def get_waffle_flag_computed_status(self, flag):
+        """
+        Produce correct "computed_status" values for WaffleFlagCourseOverrideModel instances.
+        """
+        computed_status = super().get_waffle_flag_computed_status(flag)
+        # check course overrides only if computed_status is not already 'both'
+        if computed_status != "both" and "course_overrides" in flag:
+            has_force_on = any(
+                override["force"] == "on" for override in flag["course_overrides"]
+            )
+            has_force_off = any(
+                override["force"] == "off" for override in flag["course_overrides"]
+            )
+            if has_force_on and has_force_off:
+                computed_status = "both"
+            elif has_force_on:
+                computed_status = "on" if computed_status == "on" else "both"
+            elif has_force_off:
+                computed_status = "off" if computed_status == "off" else "both"
+        return computed_status
 
 
-def _add_waffle_switch_state(switches_dict):
+class ToggleStateView(views.APIView):
     """
-    Add waffle switch state from the waffle Switch model.
+    An endpoint for displaying the state of toggles in edx-platform.
     """
-    waffle_switches = Switch.objects.all()
-    for switch_data in waffle_switches:
-        switch = _get_or_create_toggle_response(switches_dict, switch_data.name)
-        switch['is_active'] = 'true' if switch_data.active else 'false'
-        if switch_data.note:
-            switch['note'] = switch_data.note
-        switch['created'] = str(switch_data.created)
-        switch['modified'] = str(switch_data.modified)
 
+    authentication_classes = (
+        JwtAuthentication,
+        SessionAuthentication,
+    )
+    permission_classes = (
+        permissions.IsAuthenticated,
+        IsStaff,
```

Comment:
I know you are just moving existing code here, but I imagine that `IsStaff` alone will work.

Reply:
Right. I'll take the risk to remove it.